src/python/py_generate_pipeline.cpp

```diff
@@ -247,10 +278,11 @@ PYBIND11_MODULE(py_generate_pipeline, m) {
         .def_readwrite("repetition_penalty", &GenerationConfig::repetition_penalty)
         .def_readwrite("eos_token_id", &GenerationConfig::eos_token_id);
 
-    py::class_<DecodedResults>(m, "DecodedResults")
+    py::class_<DecodedResultsPyStr>(m, "DecodedResults")
         .def(py::init<>())
-        .def_readwrite("texts", &DecodedResults::texts)
-        .def_readwrite("scores", &DecodedResults::scores);
+        .def_readwrite("texts", &DecodedResultsPyStr::texts)
+        .def_readwrite("scores", &DecodedResultsPyStr::scores)
```

Comment:
why do we need to write to `DecodedResultsPyStr` ? maybe read_only ?

Reply:
I don't see a reason to prohibit writing

Reply:
I guess one reason is making an object more lightweight. Removed.